### PR TITLE
Meshworks

### PR DESCRIPTION
**Pull request by utkarsh-pro**
**Description**

This PR fixes:
1. Previous meshworks PR missed out the considerations where the hostname for the service (istio adapter) as well as meshery server can change, this PR adds environmental variables and reasonable defaults to them in both dockerfile and `main.go`
2. Previous meshworks PR missed adding the newly added `oam` directory to the github image which this PR fixes.

Adds better handling of resource deletion, that is now it deletes traits first and then proceeds to delete the application.

**Notes for Reviewers**
Please let me know if the env vars defaults are reasonable. The defaults value for Dockerfile is picked from the meshery server's docker-compose.yaml while the defaults in the main.go are in accordance with the dev setup.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,8 +21,11 @@ RUN CGO_ENABLED=1 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -ldflags="-w -
 FROM gcr.io/distroless/base
 ENV DISTRO="debian"
 ENV GOARCH="amd64"
+ENV SERVICE_ADDR="meshery-istio"
+ENV MESHERY_SERVER="http://meshery:9081"
```
Comment (leecalcote) [inline Dockerfile:25]: @kumarabd what does this mean for the adapter when deployed in a Kubernetes cluster?

Comment (kumarabd) [reply Dockerfile:25]: Just the name of the service in the grpc definition.

Comment (kumarabd) [reply Dockerfile:25]: Nothing for the kubernetes cluster

**Comment by leecalcote** [comment]
Println over to meshkit logger

**Comment by leecalcote** [comment]
Lgtm

**Review by leecalcote — APPROVED** [review]

**Pull request merged**